File: go-api/internal/modules/xm/xm.go
package xm

import (
	"net/http"
	"time"
)

// XMProject 小米运动项目
type XMProject struct {
	ID          int     `json:"id"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Price       float64 `json:"price"`
	Query       int     `json:"query"`
	Password    int     `json:"password"`
}

// XMOrder 小米运动订单
type XMOrder struct {
	ID        int         `json:"id"`
	UserID    int         `json:"user_id"`
	School    string      `json:"school"`
	Account   string      `json:"account"`
	Password  string      `json:"password"`
	ProjectID int         `json:"project_id"`
	Status    string      `json:"status_name"`
	Type      interface{} `json:"type"`
	Pace      *float64    `json:"pace"`
	Distance  *float64    `json:"distance"`
	TotalKM   int         `json:"total_km"`
	IsDeleted bool        `json:"is_deleted"`
	RunKM     *float64    `json:"run_km"`
	RunDate   interface{} `json:"run_date"`
	StartDay  string      `json:"start_day"`
	StartTime string      `json:"start_time"`
	EndTime   string      `json:"end_time"`
	Deduction float64     `json:"deduction"`
	UpdatedAt string      `json:"updated_at"`
}

type XMService struct {
	client *http.Client
}

var xmService = &XMService{
	client: &http.Client{Timeout: 15 * time.Second},
}

func XM() *XMService {
	return xmService
}

File: vben-admin/apps/web-antd/src/api/plugins/xm.ts
import { requestClient } from '#/api/request';

export interface XMProject {
  id: number;
  name: string;
  description: string;
  price: number;
  query: number;
  password: number;
}

export interface XMOrder {
  id: number;
  user_id: number;
  school: string;
  account: string;
  password: string;
  project_id: number;
  status_name: string;
  type: string | null;
  pace: number | null;
  distance: number | null;
  total_km: number;
  is_deleted: boolean;
  run_km: number | null;
  run_date: any;
  start_day: string;
  start_time: string;
  end_time: string;
  deduction: number;
  updated_at: string;
}

// 项目列表
export async function xmGetProjectsApi() {
  return requestClient.get<XMProject[]>('/xm/projects');
}

// 订单列表
export async function xmGetOrdersApi(params: {
  page: number;
  page_size: number;
  account?: string;
  school?: string;
  status?: string;
  project?: string;
  order_id?: string;
}) {
  return requestClient.get<{ list: XMOrder[]; total: number }>('/xm/orders', { params });
}

// 下单
export async function xmAddOrderApi(data: Record<string, any>) {
  return requestClient.post('/xm/add-order', data);
}

// 查询跑步状态
export async function xmQueryRunApi(data: { project_id: number; account: string; password?: string }) {
  return requestClient.post('/xm/query-run', data);
}

// 退款
export async function xmRefundOrderApi(orderId: number) {
  return requestClient.get(`/xm/refund?order_id=${orderId}`);
}

// 删除订单
export async function xmDeleteOrderApi(orderId: number) {
  return requestClient.get(`/xm/delete?order_id=${orderId}`);
}

// 同步订单
export async function xmSyncOrderApi(orderId: number) {
  return requestClient.get(`/xm/sync?order_id=${orderId}`);
}

// 增加次数/公里
export async function xmAddOrderKMApi(orderId: number, addKM: number) {
  return requestClient.post('/xm/add-order-km', { order_id: orderId, add_km: addKM });
}

// 订单日志
export async function xmGetOrderLogsApi(orderId: number, page: number, pageSize: number) {
  return requestClient.get(`/xm/order-logs?order_id=${orderId}&page=${page}&page_size=${pageSize}`);
}
